app/models/run.rb

```diff
@@ -53,11 +57,25 @@ def program(string_or_symbol)
       h[string_or_symbol.to_s]
     end
 
-    def program_from_content_type(content_type_string)
-      Run.exportable_programs.each do |prg|
-        return prg if prg.content_type == content_type_string
+    def program_from_attribute(func_name, value)
+      Run.programs.map do |prg|
+        case func_name
+        when :to_s
+          prg_value = prg.to_s
+        when :to_sym
+          prg_value = prg.to_sym
+        when :file_extension
+          prg_value = prg.file_extension
+        when :website
+          prg_value = prg.website
+        when :content_type
+          prg_value = prg.content_type
+        when :exportable
+          prg_value = prg.exportable
+        end
+        return prg if prg_value == value
       end
-      return nil
+      raise 'timer not found'
```

Comment:
oops, my previous comment was actually referring to the situation of `func_name` being unexpected, like `program_from_attribute(:blontent_type, 'application/livesplit')`. in that situation, we should `raise`.

i didn't notice that this line could be reached in an expected way, like `program_from_attribute(:content_type, 'some-new-timer')`.

we should `raise` only in the former case, the latter case should still be a `nil` return value like you had. sorry, my mistake.

Reply:
Oh, ok.  I'll change it so the default of the switch statement is a raise, and then the last line will be return nil then.  That way if it's called with an invalid attribute it throws, but otherwise it can return nil if not program is found for that attribute?